homeassistant/components/rova/sensor.py

```diff
@@ -64,39 +74,24 @@
 )
 
 
-def setup_platform(
+async def async_setup_entry(
     hass: HomeAssistant,
-    config: ConfigType,
-    add_entities: AddEntitiesCallback,
-    discovery_info: DiscoveryInfoType | None = None,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
 ) -> None:
-    """Create the Rova data service and sensors."""
-
-    zip_code = config[CONF_ZIP_CODE]
-    house_number = config[CONF_HOUSE_NUMBER]
-    house_number_suffix = config[CONF_HOUSE_NUMBER_SUFFIX]
-    platform_name = config[CONF_NAME]
-
-    # Create new Rova object to  retrieve data
-    api = Rova(zip_code, house_number, house_number_suffix)
-
-    try:
-        if not api.is_rova_area():
-            LOGGER.error("ROVA does not collect garbage in this area")
-            return
-    except (ConnectTimeout, HTTPError):
-        LOGGER.error("Could not retrieve details from ROVA API")
-        return
```

Comment:
We should instead import this yaml configuration and automatically create a new config entry from it. Check `streamlabswater` and `suez_water` for examples